broker/src/test/java/io/zeebe/broker/logstreams/AtomixLogDeletionServiceTest.java

```diff
@@ -0,0 +1,222 @@
+ */
+package io.zeebe.broker.logstreams;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.atomix.protocols.raft.storage.RaftStorage;
+import io.atomix.protocols.raft.storage.log.RaftLogReader;
+import io.atomix.storage.journal.Indexed;
+import io.atomix.storage.journal.JournalReader.Mode;
+import io.atomix.storage.journal.JournalSegmentDescriptor;
+import io.atomix.utils.time.WallClockTimestamp;
+import io.zeebe.broker.clustering.atomix.storage.snapshot.AtomixRecordEntrySupplierImpl;
+import io.zeebe.broker.clustering.atomix.storage.snapshot.AtomixSnapshotStorage;
+import io.zeebe.broker.clustering.atomix.storage.snapshot.DbSnapshotStore;
+import io.zeebe.logstreams.state.SnapshotMetrics;
+import io.zeebe.logstreams.storage.atomix.AtomixLogStorageReader;
+import io.zeebe.logstreams.util.AtomixLogStorageRule;
+import io.zeebe.util.sched.testing.ActorSchedulerRule;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.ByteBuffer;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.concurrent.TimeUnit;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TemporaryFolder;
+
+public final class AtomixLogDeletionServiceTest {
+
+  private static final ByteBuffer DATA = ByteBuffer.allocate(Integer.BYTES).putInt(0, 1);
+  private static final int PARTITION_ID = 1;
+
+  private final ActorSchedulerRule actorScheduler = new ActorSchedulerRule();
+  private final TemporaryFolder temporaryFolder = new TemporaryFolder();
+  private final AtomixLogStorageRule logStorageRule =
+      new AtomixLogStorageRule(temporaryFolder, PARTITION_ID, b -> builder(b, temporaryFolder));
+
+  @Rule
+  public final RuleChain chain =
+      RuleChain.outerRule(temporaryFolder).around(actorScheduler).around(logStorageRule);
+
+  private AtomixSnapshotStorage snapshotStorage;
+  private AtomixLogStorageReader storageReader;
+  private LogDeletionService deletionService;
+  private Compactor compactor;
+
+  @Before
+  public void setUp() {
+    storageReader =
+        new AtomixLogStorageReader(logStorageRule.getRaftLog().openReader(-1, Mode.COMMITS));
+    snapshotStorage =
+        new AtomixSnapshotStorage(
+            null,
+            logStorageRule.getSnapshotStore(),
+            new AtomixRecordEntrySupplierImpl(storageReader),
+            1,
+            new SnapshotMetrics(PARTITION_ID));
+    compactor = new Compactor();
+
+    deletionService = new LogDeletionService(0, PARTITION_ID, compactor, snapshotStorage);
+    actorScheduler.submitActor(deletionService).join();
+  }
+
+  @After
+  public void tearDown() {
+    if (storageReader != null) {
```

Comment:
Why do you think they are null?

Reply:
Force of habit, in case something went wrong before it was initialized and the `@After` still runs - just defensive programming in general.

Reply:
Hm not sure if this make sense here, it is just noise on reading the test.